src/semaphore.c

```diff
@@ -474,6 +507,11 @@ _dispatch_group_wake(dispatch_semaphore_t dsema)
 			int ret = sem_post(&dsema->dsema_sem);
 			DISPATCH_SEMAPHORE_VERIFY_RET(ret);
 		} while (--rval);
+#elif USE_FUTEX_SEM
+		do {
+			int ret = _dispatch_futex_signal(&dsema->dsema_futex);
```

Comment:
actually now that I read the full patch I understand why, but it feels sad that FUTEX_WAKE with `n` can't be use, I assume it's not the end of the world though.